userapi/src/main/java/controller/RouterVerticle.java

```diff
@@ -0,0 +1,148 @@
+package controller;
+
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.vertx.core.AbstractVerticle;
+import io.vertx.core.CompositeFuture;
+import io.vertx.core.Future;
+import io.vertx.core.http.HttpServer;
+import io.vertx.core.json.Json;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.ext.web.handler.TimeoutHandler;
+import model.D42Model;
+import model.IDataModel;
+import model.Profile;
+
+import java.util.Set;
+
+/**
+ * Routes requests to their respective handlers
+ * Created by rohit.patiyal on 18/01/17.
+ */
+public class RouterVerticle extends AbstractVerticle {
+
+    private static final long REQ_TIMEOUT = 2000;
+    private IDataModel dataModel = null;
+
+    private void setIDataModel(IDataModel IDataModel) {
+        if (dataModel == null) {
+            this.dataModel = IDataModel;
+        }
+    }
+
+    private Router configureRouter() {
+        Router router = Router.router(vertx);
+        router.route().handler(TimeoutHandler.create(REQ_TIMEOUT));
+        router.route("/").handler(routingContext -> routingContext.response()
+                .putHeader("context-type", "text/html")
+                .end("<h1>Welcome to UserAPI for FKProfiler"));
+        router.get("/apps").blockingHandler(this::getAppIds, false);
+        router.get("/cluster/:appId").blockingHandler(this::getClusterIds, false);
+        router.get("/proc/:appId/:clusterId").blockingHandler(this::getProcs, false);
+        router.get("/profiles/:appId/:clusterId/:proc").blockingHandler(this::getProfiles, false);
+        // router.get("/traces/:appId/:clusterId/:proc/:workType").blockingHandler(this.getTraces, false);
+
+        return router;
+    }
+
+    @Override
+    public void start(Future<Void> startFuture) throws Exception {
+        Router router = configureRouter();
+
+        Future<HttpServer> serverFuture = Future.future();
+        Future<String> dbFuture = Future.future();
+
+        vertx.createHttpServer()
+                .requestHandler(router::accept)
+                .listen(config().getInteger("http.port", 8080), serverFuture.completer());
+
+
+        vertx.executeBlocking(future -> {
+            setIDataModel(new D42Model());
```

Comment:
The dependency on the interface will be required because it is actually the interface (IDataModel class, name to be refactored) which is making the call to the storage listAsync API. Although moving the client initialisation will make this class independent of the DataModel implementation.